File: max_value/max_value.py
def max_value(nums):
    """_summary_
    Finds the maximum value in a list of numbers.

    Args:
        nums (_type_):  A list of numbers.



    Returns:
        _type_: _description_
        The maximum value in the list. If the list is empty, it returns float('-inf').

    Complexity:
        Time: O(n)
        Space: O(1)

    Steps:
    1. Initialize a variable `minimum_value` with the smallest possible value in Python, which is float('-inf').
    2. Iterate through the list of numbers.
    3. If the current number is greater than `minimum_value`, update `minimum_value` with the current number.
    4. After iterating through the entire list, return `minimum_value`.



    """
    minimum_value = float("-inf")
    for num in nums:
        if num > minimum_value:
            minimum_value = num
    return minimum_value


print(max_value([3, 5, 1, 7, 2]))  # 7
print(max_value([10, 20, 30]))  # 30
print(max_value([]))  # float('-inf')

File: max_value/max_value.js
/**
 * Write a function, maxValue, that takes in array of numbers as an argument. The function should return the largest number in the array.

Solve this without using any built-in array methods.

You can assume that the array is non-empty.
 */

/**
 * maxValue - function to get the maximum value in an array of numbers.
 * @param {array} nums - array of numbers
 * returns the maximum value in the array
 */

const maxValue = (nums) => {
  // Steps:
  // 1. create a variable to store the minimum value
  // 2. loop through the numbers array
  // 3. check if the current number is greater than the minimum value
  // 4. if it is, update the minimum value
  // 5. return the minimum value
  let minimumValue = -Infinity;

  for (let num of nums) {
    if (num > minimumValue) {
      minimumValue = num;
    }
  }
  //   for (let i = 0; i < nums.length; i++) {
  //     if (nums[i] > minimumValue) {
  //       minimumValue = nums[i];
  //     }
  //   }
  return minimumValue;
};

console.log(maxValue([4, 5, 1, 3])); // 5
console.log(maxValue([10, 20, 30, 40])); // 40
console.log(maxValue([100, 50, 200])); // 200
console.log(maxValue([1])); // 1
console.log(maxValue([3, 2, 1])); // 3
console.log(maxValue([0, -10, -20, -30])); // 0
